parse.go

```diff
@@ -694,20 +748,39 @@ func isFlag(s string) bool {
 	return strings.HasPrefix(s, "-") && strings.TrimLeft(s, "-") != ""
 }
 
-// val returns a reflect.Value corresponding to the current value for the
-// given path
-func (p *Parser) val(dest path) reflect.Value {
+// getVal returns a reflect.Value corresponding to the current value for
+// the given path, but does not instantiate nil values in the path.
+func (p *Parser) getVal(dest path) reflect.Value {
```

Comment:
perhaps call this `mustVal` to indicate that it assumes that the path is initialized already?

Reply:
The `getVal(...)` function was really only used in 1 place where the updated behavior of `val(...)` (to instantiate nil-pointer parents in the path) was not needed. This function now returns the string representation of any non-default values encountered in the parer struct. It can return early when nil-pointers are detected in the path since there cannot be any non-default values in a non-existent child element.